src/main/java/core/Storage.java

```diff
@@ -0,0 +1,100 @@
+package core;
+
+import dukexception.DukeException;
+import dukexception.StorageException;
+import task.Task;
+
+import java.io.*;
+import java.util.ArrayList;
+
+public class Storage {
+    private static final String fileName= "duke.txt";
+    private File file;
+    private DataParser dataParser;
+
+    /**
+     * Constructor to initialize the path of the external storage.
+     */
+    public Storage(){
+        file=new File(fileName);
+        dataParser=new DataParser();
+    }
+
+    /**
+     * Saves the states of the system to external file.
+     * @param tasks is the states of the system.
+     * @throws DukeException when writing to file is unsuccessful.
+     */
+    public void save(ArrayList<Task> tasks) throws DukeException {
+        writeTo(file,tasks);
+    }
+
+    /**
+     * Loads the states from the external file to the system.
+     * @return the states of the system.
+     * @throws DukeException when reading from the file is unsuccessful.
+     */
+    public ArrayList<Task> load() throws DukeException{
+        return readFrom(file);
+    }
+
+    /**
+     * Clears all the data in the external file.
+     * @throws DukeException when reading from the file is unsuccessful.
+     */
+    public void clearData() throws DukeException{
+        try(FileWriter fw=new FileWriter(file)){
+            BufferedWriter bw=new BufferedWriter(fw);
+            bw.close();
+        }catch(IOException ex){
+            throw new DukeException("Encounter error in resetting data");
+        }
+    }
+
+    /**
+     * Obtains the task lsit from the external file.
+     * @param thisFile indicates the file to be read from.
+     * @return the task list.
+     * @throws DukeException when reading from the file is unsuccessful.
+     */
+    private ArrayList<Task> readFrom(File thisFile) throws DukeException{
+        ArrayList<Task> tasks=new ArrayList<>();
+        try(FileReader fr=new FileReader(thisFile)) {
+            BufferedReader br=new BufferedReader(fr);
+            String line;
+            while ((line = br.readLine()) != null) {
+                Task task=dataParser.parseToTask(line);
+                tasks.add(task);
+            }
+            br.close();
```

Comment:
Maybe you can put the br.close() in a finally block to ensure you always close it